patterns/shared-nullifier-key/src/main.nr

```diff
@@ -0,0 +1,123 @@
+mod types;
+ 
+contract SharedNullifierKey { 
+    use crate::types::shared_note:: {SharedNote, SharedNoteMethods, SHARED_NOTE_LEN};
+
+    use dep::std::option::Option;
+
+    use dep::aztec::context::Context;
+    use dep::aztec::log::emit_encrypted_log;
+    
+    use dep::aztec::note::{
+        note_getter_options::NoteGetterOptions,
+        note_header::NoteHeader,
+        note_interface::NoteInterface,
+        note_viewer_options::NoteViewerOptions,
+        utils as note_utils
+    };
+
+    use dep::aztec::oracle::{
+        get_public_key::get_public_key,
+        rand::rand,
+    };
+    use dep::aztec::state_vars::{
+        set::Set,
+    };
+
+    use dep::aztec::protocol_types::{
+        address::AztecAddress,
+    };
```

Comment:
nit> just realized this: maybe tidier to group by aztec imports, wdyt?

Reply:
yes, agreed, created a task to do it at the end